### DISCUSSION TITLE
what docker tag to use to test 0.15 / gen-ai and notifications

### DISCUSSION THREAD

**Discussion by MrNews89** [blakeblackshear/frigate#13597] [General]
Hi

I am peeking into the dev docs and i am totally exited to try out the new features. I would love to spin up a test instance with gen-ai and notifications. I am a noob in terms of github though, and request if anyone can direct me how to a docker image compiled with the above features?

**Comment by NickM-27** [comment]
dev builds are made automatically, just use the latest commit hash https://github.com/users/blakeblackshear/packages/container/package/frigate

**Comment by hawkeye217** [comment]
It should also be noted for noobs that the dev builds are still very much a work in progress and don't represent a final release product. If you choose to run them, run them at your own risk. Expect changes (and sometimes pretty nasty bugs) from build to build.

**Comment by MrNews89** [comment]
thank you both. i plan to run it in a separate instance with separate media and config.

**Comment by jameslivulpi** [comment]
Been using it a few days with gemini flash.. Love this new feature. 

<img width="571" alt="Screenshot 2024-09-07 at 9 40 14 PM" src="https://github.com/user-attachments/assets/33c0f6e2-0a7c-4535-ba6b-361c35acd727">

so far about 5900 calls to gemini endpoint and working flawlessly only enabled on 1 camera

**Reply by divemasterjm** [reply]
wich tag do you use?

**Reply by jameslivulpi** [reply]
Currently I am using the latest commit and update pretty often:
https://github.com/blakeblackshear/frigate/commits/dev/

so image would be:

```
image: ghcr.io/blakeblackshear/frigate:17901fc
```

keep in mind this 
> It should also be noted for noobs that the dev builds are still very much a work in progress and don't represent a final release product. If you choose to run them, run them at your own risk. Expect changes (and sometimes pretty nasty bugs) from build to build.

https://github.com/blakeblackshear/frigate/discussions/13597#discussioncomment-10573880

**Reply by divemasterjm** [reply]
> Currently I am using the latest commit and update pretty often: https://github.com/blakeblackshear/frigate/commits/dev/
> 
> so image would be:
> 
> ```
> image: ghcr.io/blakeblackshear/frigate:17901fc
> ```
> 
> keep in mind this
> 
> > It should also be noted for noobs that the dev builds are still very much a work in progress and don't represent a final release product. If you choose to run them, run them at your own risk. Expect changes (and sometimes pretty nasty bugs) from build to build.
> 
> [#13597 (comment)](https://github.com/blakeblackshear/frigate/discussions/13597#discussioncomment-10573880)

thanks, i was looking in packages and there are 014 and 015.

**Comment by DrSpaldo** [comment]
I usually play around and test the dev builds. In the last few the docs link appears to just go to the standard docs. Is there a way to work out by using the build number or something where the dev docs are?

**Reply by DrSpaldo** [reply]
I think I've worked it out and a way to find them. If you go into some (not all) of the pull requests, they have a docs link :)

**Reply by NickM-27** [reply]
you can just use the 0.15 PR https://github.com/blakeblackshear/frigate/pull/13787

**Reply by dlong500** [reply]
@NickM-27 I tried looking around and maybe missed something, but has there been anything mentioned in regards to an approximate release window for 0.15?  No pressure, just curious...

**Reply by hawkeye217** [reply]
There's no approximate release date available at this time. There's still bugs we're fixing and features we're developing. It'll be ready when it's ready :)

**Reply by DrSpaldo** [reply]
@dlong500 just give one of the dev builds a go. They aren't very buggy usually.... 

ie. ghcr.io/blakeblackshear/frigate:3c591ad

Full list https://github.com/blakeblackshear/frigate/pkgs/container/frigate

**Reply by leccelecce** [reply]
I think I looked before, but is there a roadmap showing expected feature list for 0.15 (or indeed any subsequent releases)? I've seen various potential features mentioned around the place, but it would be great to have a high level list.

For example, those of us who keep up with the project have a good idea what's coming in 0.15 (Further UI improvements, semantic search etc), and maybe even some changes deferred to 0.16, but without anyone else scrolling through hundreds and hundreds of PRs it's difficult to know what's coming.

Fully appreciate the contributor-based model of development, if anything I think it would help reduce questions and continue building the excitement around the project :-)

**Reply by DrSpaldo** [reply]
@leccelecce lots of posts asking about roadmaps, they usually say it is just in their head. This is probably the best you will get for the 0.15 release:

https://github.com/blakeblackshear/frigate/pull/13787

**Reply by leccelecce** [reply]
Yeah, I'm aware of the PR and have a good idea what's coming (so far) in 0.15 - it was more about what might be coming in 0.16, 0.17, 0.18, and equally the idea that publishing a roadmap (or the fact that there isn't a public one) would save questions.

For example, fine-grained user roles have been mentioned. I would like to open up a couple of Frigate instances I look after, but not until this is done. Currently not sure if that's coming in 3 months, 6 months, 12 months...I suspect that's a feature quite a few folks are interested in (and a determining factor for certain use cases outside homes)

**Reply by NickM-27** [reply]
There's no way to know a time frame of when a feature is coming, we do this in our free time so there's no telling how much time we will have for working on stuff. 

User roles is a high priority feature that we've already discussed how we'd want to implement it.

**Reply by hawkeye217** [reply]
As one of the larger contributors to Frigate, I can say that we really don't have a "roadmap" in the traditional sense - at least certainly not enough to publish anything. As we've said before elsewhere, we do have some of the major features we're looking to implement in mind. But because we're just volunteers on Blake's "weekend project" with our own jobs, families, and hobbies, we have no set schedule on when we plan to release anything.

But the beauty of open source is that development is not just limited to the few of us main contributors. We'd encourage anyone in the community that may be interested in contributing to talk to us and then begin writing code!

**Reply by leccelecce** [reply]
The above (or a slightly tweaked version) would be great to add to the project readme under a "Roadmap" heading!
